### ISSUE TITLE
Browse Button crashes after adding new field to a table

### ISSUE THREAD

**Issue by louieuow** [phpmyadmin/phpmyadmin#16564]
### Describe the bug

All was working fine with the view of the table contents before adding the bottom field
I added a field to the bottom of the table through the structure button. When I clicked the Browse button a series of error messages appeared, instead of the database contents.

I dropped the added the field and clicked the  browse button and the database contents were displayed as normal.
I re-added the field and the error came back again after clicking the Browse button

I repeated the dropping and adding of the field from the server side rather than use PHPMyAdmin and the error still occurred in PHPMyAdmin when the field was re-added 

### To Reproduce

Added field using the Structure button and all looked normal
clicked on Browse button and  error messages where displayed

### Expected behavior

After clicking the Browse button the database contents should appear

### Screenshots

The 'structure.PNG' shows the table structure after adding the field number 8 - result_count
![image](https://user-images.githubusercontent.com/1317389/104254872-e9a60d00-54cb-11eb-863f-8457a8eb39e8.png)

The 'browse button error.PNG' shows the error message after adding the field and clicking Browse button
![image](https://user-images.githubusercontent.com/1317389/104254931-03dfeb00-54cc-11eb-8a6a-3afe3a0b2b00.png)

The 'after field removed.PNG' shows the normal output after the added  field was removed
![image](https://user-images.githubusercontent.com/1317389/104255144-81a3f680-54cc-11eb-8931-486c9dcb4913.png)

**_NOTE:_**
I have noticed that the field order in the third screen shot is different to the field order in the table structure in first screen shot, could this be the cause of the issue?

### Server configuration

- Operating system: Windows Server 2012
- Web server: Apache 2.4.46
- Database version: mysql  Ver 14.14 Distrib 5.6.14, for Win64 (x86_64)
- PHP version:  7.4.13
- phpMyAdmin version: 5.02 and 5.04

### Client configuration

- Browser: All browsers -  Chrome, Edge, Firefox
- Operating system:  Windows 10 Enterprise 

### Additional context

Output in the error screen after clinking the Browse Button
```
Notice in .\libraries\classes\Display\Results.php#1086
Undefined offset: 7

Backtrace

Results.php#1220: PhpMyAdmin\Display\Results->_getTableHeadersForColumns(
array,
array,
array,
array,
array,
boolean false,
string 'SELECT * FROM `results_session` ',
)
Results.php#4299: PhpMyAdmin\Display\Results->_getTableHeaders(
array,
array,
string 'SELECT * FROM `results_session` ',
array,
array,
array,
boolean false,
)
Sql.php#1669: PhpMyAdmin\Display\Results->getTable(
,
array,
array,
boolean false,
)
Sql.php#2007: PhpMyAdmin\Sql->getHtmlForSqlQueryResultsTable(
,
string './themes/pmahomme/img/',
NULL,
array,
boolean true,
string '403',
integer 25,
NULL,
,
array,
)
Sql.php#2276: PhpMyAdmin\Sql->getQueryResponseForResultsReturned(
,
array,
string 'central',
string 'results_session',
NULL,
NULL,
,
string './themes/pmahomme/img/',
string '403',
integer 25,
NULL,
NULL,
NULL,
NULL,
NULL,
string 'SELECT * FROM `results_session`',
NULL,
)
Sql.php#2105: PhpMyAdmin\Sql->executeQueryAndGetQueryResponse(
array,
boolean true,
string 'central',
string 'results_session',
NULL,
NULL,
NULL,
NULL,
NULL,
NULL,
string '',
string './themes/pmahomme/img/',
NULL,
NULL,
NULL,
string 'SELECT * FROM `results_session`',
NULL,
NULL,
)
sql.php#242: PhpMyAdmin\Sql->executeQueryAndSendQueryResponse(
array,
boolean true,
string 'central',
string 'results_session',
NULL,
NULL,
NULL,
NULL,
NULL,
NULL,
string '',
string './themes/pmahomme/img/',
NULL,
NULL,
NULL,
string 'SELECT * FROM `results_session`',
NULL,
NULL,
)
Notice in .\libraries\classes\Display\Results.php#1090
Undefined index:

Backtrace

Results.php#1220: PhpMyAdmin\Display\Results->_getTableHeadersForColumns(
array,
array,
array,
array,
array,
boolean false,
string 'SELECT * FROM `results_session` ',
)
Results.php#4299: PhpMyAdmin\Display\Results->_getTableHeaders(
array,
array,
string 'SELECT * FROM `results_session` ',
array,
array,
array,
boolean false,
)
Sql.php#1669: PhpMyAdmin\Display\Results->getTable(
,
array,
array,
boolean false,
)
Sql.php#2007: PhpMyAdmin\Sql->getHtmlForSqlQueryResultsTable(
,
string './themes/pmahomme/img/',
NULL,
array,
boolean true,
string '403',
integer 25,
NULL,
,
array,
)
Sql.php#2276: PhpMyAdmin\Sql->getQueryResponseForResultsReturned(
,
array,
string 'central',
string 'results_session',
NULL,
NULL,
,
string './themes/pmahomme/img/',
string '403',
integer 25,
NULL,
NULL,
NULL,
NULL,
NULL,
string 'SELECT * FROM `results_session`',
NULL,
)
Sql.php#2105: PhpMyAdmin\Sql->executeQueryAndGetQueryResponse(
array,
boolean true,
string 'central',
string 'results_session',
NULL,
NULL,
NULL,
NULL,
NULL,
NULL,
string '',
string './themes/pmahomme/img/',
NULL,
NULL,
NULL,
string 'SELECT * FROM `results_session`',
NULL,
NULL,
)
sql.php#242: PhpMyAdmin\Sql->executeQueryAndSendQueryResponse(
array,
boolean true,
string 'central',
string 'results_session',
NULL,
NULL,
NULL,
NULL,
NULL,
NULL,
string '',
string './themes/pmahomme/img/',
NULL,
NULL,
NULL,
string 'SELECT * FROM `results_session`',
NULL,
NULL,
)
Notice in .\libraries\classes\Display\Results.php#1090
Trying to get property 'name' of non-object

Backtrace

Results.php#1220: PhpMyAdmin\Display\Results->_getTableHeadersForColumns(
array,
array,
array,
array,
array,
boolean false,
string 'SELECT * FROM `results_session` ',
)
Results.php#4299: PhpMyAdmin\Display\Results->_getTableHeaders(
array,
array,
string 'SELECT * FROM `results_session` ',
array,
array,
array,
boolean false,
)
Sql.php#1669: PhpMyAdmin\Display\Results->getTable(
,
array,
array,
boolean false,
)
Sql.php#2007: PhpMyAdmin\Sql->getHtmlForSqlQueryResultsTable(
,
string './themes/pmahomme/img/',
NULL,
array,
boolean true,
string '403',
integer 25,
NULL,
,
array,
)
Sql.php#2276: PhpMyAdmin\Sql->getQueryResponseForResultsReturned(
,
array,
string 'central',
string 'results_session',
NULL,
NULL,
,
string './themes/pmahomme/img/',
string '403',
integer 25,
NULL,
NULL,
NULL,
NULL,
NULL,
string 'SELECT * FROM `results_session`',
NULL,
)
Sql.php#2105: PhpMyAdmin\Sql->executeQueryAndGetQueryResponse(
array,
boolean true,
string 'central',
string 'results_session',
NULL,
NULL,
NULL,
NULL,
NULL,
NULL,
string '',
string './themes/pmahomme/img/',
NULL,
NULL,
NULL,
string 'SELECT * FROM `results_session`',
NULL,
NULL,
)
sql.php#242: PhpMyAdmin\Sql->executeQueryAndSendQueryResponse(
array,
boolean true,
string 'central',
string 'results_session',
NULL,
NULL,
NULL,
NULL,
NULL,
NULL,
string '',
string './themes/pmahomme/img/',
NULL,
NULL,
NULL,
string 'SELECT * FROM `results_session`',
NULL,
NULL,
)
Notice in .\libraries\classes\Display\Results.php#1097
Undefined index:

Backtrace

Results.php#1220: PhpMyAdmin\Display\Results->_getTableHeadersForColumns(
array,
array,
array,
array,
array,
boolean false,
string 'SELECT * FROM `results_session` ',
)
Results.php#4299: PhpMyAdmin\Display\Results->_getTableHeaders(
array,
array,
string 'SELECT * FROM `results_session` ',
array,
array,
array,
boolean false,
)
Sql.php#1669: PhpMyAdmin\Display\Results->getTable(
,
array,
array,
boolean false,
)
Sql.php#2007: PhpMyAdmin\Sql->getHtmlForSqlQueryResultsTable(
,
string './themes/pmahomme/img/',
NULL,
array,
boolean true,
string '403',
integer 25,
NULL,
,
array,
)
Sql.php#2276: PhpMyAdmin\Sql->getQueryResponseForResultsReturned(
,
array,
string 'central',
string 'results_session',
NULL,
NULL,
,
string './themes/pmahomme/img/',
string '403',
integer 25,
NULL,
NULL,
NULL,
NULL,
NULL,
string 'SELECT * FROM `results_session`',
NULL,
)
Sql.php#2105: PhpMyAdmin\Sql->executeQueryAndGetQueryResponse(
array,
boolean true,
string 'central',
string 'results_session',
NULL,
NULL,
NULL,
NULL,
NULL,
NULL,
string '',
string './themes/pmahomme/img/',
NULL,
NULL,
NULL,
string 'SELECT * FROM `results_session`',
NULL,
NULL,
)
sql.php#242: PhpMyAdmin\Sql->executeQueryAndSendQueryResponse(
array,
boolean true,
string 'central',
string 'results_session',
NULL,
NULL,
NULL,
NULL,
NULL,
NULL,
string '',
string './themes/pmahomme/img/',
NULL,
NULL,
NULL,
string 'SELECT * FROM `results_session`',
NULL,
NULL,
)
Notice in .\libraries\classes\Display\Results.php#1103
Undefined index:

Backtrace

Results.php#1220: PhpMyAdmin\Display\Results->_getTableHeadersForColumns(
array,
array,
array,
array,
array,
boolean false,
string 'SELECT * FROM `results_session` ',
)
Results.php#4299: PhpMyAdmin\Display\Results->_getTableHeaders(
array,
array,
string 'SELECT * FROM `results_session` ',
array,
array,
array,
boolean false,
)
Sql.php#1669: PhpMyAdmin\Display\Results->getTable(
,
array,
array,
boolean false,
)
Sql.php#2007: PhpMyAdmin\Sql->getHtmlForSqlQueryResultsTable(
,
string './themes/pmahomme/img/',
NULL,
array,
boolean true,
string '403',
integer 25,
NULL,
,
array,
)
Sql.php#2276: PhpMyAdmin\Sql->getQueryResponseForResultsReturned(
,
array,
string 'central',
string 'results_session',
NULL,
NULL,
,
string './themes/pmahomme/img/',
string '403',
integer 25,
NULL,
NULL,
NULL,
NULL,
NULL,
string 'SELECT * FROM `results_session`',
NULL,
)
Sql.php#2105: PhpMyAdmin\Sql->executeQueryAndGetQueryResponse(
array,
boolean true,
string 'central',
string 'results_session',
NULL,
NULL,
NULL,
NULL,
NULL,
NULL,
string '',
string './themes/pmahomme/img/',
NULL,
NULL,
NULL,
string 'SELECT * FROM `results_session`',
NULL,
NULL,
)
sql.php#242: PhpMyAdmin\Sql->executeQueryAndSendQueryResponse(
array,
boolean true,
string 'central',
string 'results_session',
NULL,
NULL,
NULL,
NULL,
NULL,
NULL,
string '',
string './themes/pmahomme/img/',
NULL,
NULL,
NULL,
string 'SELECT * FROM `results_session`',
NULL,
NULL,
)
Notice in .\libraries\classes\Display\Results.php#1689
Trying to get property 'name' of non-object

Backtrace

Results.php#1103: PhpMyAdmin\Display\Results->_getOrderLinkAndSortedHeaderHtml(
NULL,
array,
array,
NULL,
string 'SELECT * FROM `results_session` ',
integer 25,
string '',
array,
array,
NULL,
)
Results.php#1220: PhpMyAdmin\Display\Results->_getTableHeadersForColumns(
array,
array,
array,
array,
array,
boolean false,
string 'SELECT * FROM `results_session` ',
)
Results.php#4299: PhpMyAdmin\Display\Results->_getTableHeaders(
array,
array,
string 'SELECT * FROM `results_session` ',
array,
array,
array,
boolean false,
)
Sql.php#1669: PhpMyAdmin\Display\Results->getTable(
,
array,
array,
boolean false,
)
Sql.php#2007: PhpMyAdmin\Sql->getHtmlForSqlQueryResultsTable(
,
string './themes/pmahomme/img/',
NULL,
array,
boolean true,
string '403',
integer 25,
NULL,
,
array,
)
Sql.php#2276: PhpMyAdmin\Sql->getQueryResponseForResultsReturned(
,
array,
string 'central',
string 'results_session',
NULL,
NULL,
,
string './themes/pmahomme/img/',
string '403',
integer 25,
NULL,
NULL,
NULL,
NULL,
NULL,
string 'SELECT * FROM `results_session`',
NULL,
)
Sql.php#2105: PhpMyAdmin\Sql->executeQueryAndGetQueryResponse(
array,
boolean true,
string 'central',
string 'results_session',
NULL,
NULL,
NULL,
NULL,
NULL,
NULL,
string '',
string './themes/pmahomme/img/',
NULL,
NULL,
NULL,
string 'SELECT * FROM `results_session`',
NULL,
NULL,
)
sql.php#242: PhpMyAdmin\Sql->executeQueryAndSendQueryResponse(
array,
boolean true,
string 'central',
string 'results_session',
NULL,
NULL,
NULL,
NULL,
NULL,
NULL,
string '',
string './themes/pmahomme/img/',
NULL,
NULL,
NULL,
string 'SELECT * FROM `results_session`',
NULL,
NULL,
)
Notice in .\libraries\classes\Display\Results.php#2065
Trying to get property 'name' of non-object

Backtrace

Results.php#1739: PhpMyAdmin\Display\Results->_getSortOrderLink(
string '',
NULL,
string 'sql.php?db=central&amp;table=results_session&amp;sql_query=SELECT+%2A+FROM+%60results_session%60++&amp;sql_signature=393a25d83d3440e79dca537c5931f3a4e4513989c45b3dc199b3375888ce6db5&amp;session_max_rows=25&amp;is_browse_distinct=0',
string 'sql.php?db=central&amp;table=results_session&amp;sql_query=SELECT+%2A+FROM+%60results_session%60++&amp;sql_signature=393a25d83d3440e79dca537c5931f3a4e4513989c45b3dc199b3375888ce6db5&amp;session_max_rows=25&amp;is_browse_distinct=0',
)
Results.php#1103: PhpMyAdmin\Display\Results->_getOrderLinkAndSortedHeaderHtml(
NULL,
array,
array,
NULL,
string 'SELECT * FROM `results_session` ',
integer 25,
string '',
array,
array,
NULL,
)
Results.php#1220: PhpMyAdmin\Display\Results->_getTableHeadersForColumns(
array,
array,
array,
array,
array,
boolean false,
string 'SELECT * FROM `results_session` ',
)
Results.php#4299: PhpMyAdmin\Display\Results->_getTableHeaders(
array,
array,
string 'SELECT * FROM `results_session` ',
array,
array,
array,
boolean false,
)
Sql.php#1669: PhpMyAdmin\Display\Results->getTable(
,
array,
array,
boolean false,
)
Sql.php#2007: PhpMyAdmin\Sql->getHtmlForSqlQueryResultsTable(
,
string './themes/pmahomme/img/',
NULL,
array,
boolean true,
string '403',
integer 25,
NULL,
,
array,
)
Sql.php#2276: PhpMyAdmin\Sql->getQueryResponseForResultsReturned(
,
array,
string 'central',
string 'results_session',
NULL,
NULL,
,
string './themes/pmahomme/img/',
string '403',
integer 25,
NULL,
NULL,
NULL,
NULL,
NULL,
string 'SELECT * FROM `results_session`',
NULL,
)
Sql.php#2105: PhpMyAdmin\Sql->executeQueryAndGetQueryResponse(
array,
boolean true,
string 'central',
string 'results_session',
NULL,
NULL,
NULL,
NULL,
NULL,
NULL,
string '',
string './themes/pmahomme/img/',
NULL,
NULL,
NULL,
string 'SELECT * FROM `results_session`',
NULL,
NULL,
)
sql.php#242: PhpMyAdmin\Sql->executeQueryAndSendQueryResponse(
array,
boolean true,
string 'central',
string 'results_session',
NULL,
NULL,
NULL,
NULL,
NULL,
NULL,
string '',
string './themes/pmahomme/img/',
NULL,
NULL,
NULL,
string 'SELECT * FROM `results_session`',
NULL,
NULL,
)

```

**Comment by williamdes** [comment]
Hi @louieuow 
Thanks for reporting this issue and sorry for the response delay

Could you try to reproduce this issue on one of the Demo servers (`root`, no password) ?

- [`5.2.0-dev`](https://demo.phpmyadmin.net/master/)
- [`5.1.x-dev`](https://demo.phpmyadmin.net/QA_5_1/)
- [`5.0.x-dev`](https://demo.phpmyadmin.net/QA_5_0/)
- [`4.9.x`](https://demo.phpmyadmin.net/QA_4_9/)

If you can please export the structure and data here because the server is emptied by a cron.

**Comment by louieuow** [comment]
The database that contains the table causing the error, contains 61 tables and is over 600Mb in data so not possible to replicate data in your environment. I tried importing the whole database schema without data to the 4.9.x and 5.2.0-dev  sites with no issues. 
I then imported the offending table and its data, removed and re-added the result_count field. There is was no errors on your PHPMyAdmin servers. 

I even installed 5.1 RC1 on my server and the issue is still there.

The issue appears to be local between my Database server and PHPMyAdmin and how the field was added to the database schema.

I then tried the following to correct the issue.--->

I export the  "results_session" table and data  with PHPMyAdmin  Export, with 'Drop Table' command to the screen.

I manually dropped the "results_session" table first.

I copied the export script to the SQL tab and ran it. 

Somehow this corrected the issue an PHPMyAdmin, and I can now browse the database, add and remove fields a browse. Even through there was no issue with browsing the table with the MySQL GUI clients and command line.

The issue appears to have been some freak event when the table was first created and resolved when the table was dropped and reinstated.
